packages/eservice-template-process/src/model/domain/toEvent.ts

```diff
@@ -0,0 +1,472 @@
+/* eslint-disable max-params */
+import { CreateEvent } from "pagopa-interop-commons";
+import {
+  EServiceTemplateEvent,
+  toEServiceTemplateV2,
+  CorrelationId,
+  EServiceTemplate,
+  RiskAnalysisId,
+  EServiceTemplateVersionId,
+  EServiceDocumentId,
+  AttributeId,
+} from "pagopa-interop-models";
+
+export const toCreateEventEServiceTemplateAdded = (
+  eserviceTemplate: EServiceTemplate,
+  correlationId: CorrelationId
+): CreateEvent<EServiceTemplateEvent> => ({
+  streamId: eserviceTemplate.id,
+  version: 0,
+  event: {
+    type: "EServiceTemplateAdded",
+    event_version: 2,
+    data: { eserviceTemplate: toEServiceTemplateV2(eserviceTemplate) },
+  },
+  correlationId,
+});
+
+export const toCreateEventEServiceTemplateRiskAnalysisAdded = (
+  streamId: string,
+  version: number,
+  riskAnalysisId: RiskAnalysisId,
+  eserviceTemplate: EServiceTemplate,
+  correlationId: CorrelationId
+): CreateEvent<EServiceTemplateEvent> => ({
+  streamId,
+  version,
+  event: {
+    type: "EServiceTemplateRiskAnalysisAdded",
+    event_version: 2,
+    data: {
+      riskAnalysisId,
+      eserviceTemplate: toEServiceTemplateV2(eserviceTemplate),
+    },
+  },
+  correlationId,
+});
+
+export const toCreateEventEServiceTemplateRiskAnalysisDeleted = (
+  streamId: string,
+  version: number,
+  riskAnalysisId: RiskAnalysisId,
+  eserviceTemplate: EServiceTemplate,
+  correlationId: CorrelationId
+): CreateEvent<EServiceTemplateEvent> => ({
+  streamId,
+  version,
+  event: {
+    type: "EServiceTemplateRiskAnalysisDeleted",
+    event_version: 2,
+    data: {
+      riskAnalysisId,
+      eserviceTemplate: toEServiceTemplateV2(eserviceTemplate),
+    },
+  },
+  correlationId,
+});
+
+export const toCreateEventEServiceTemplateRiskAnalysisUpdated = (
+  streamId: string,
+  version: number,
+  riskAnalysisId: RiskAnalysisId,
+  eserviceTemplate: EServiceTemplate,
+  correlationId: CorrelationId
+): CreateEvent<EServiceTemplateEvent> => ({
+  streamId,
+  version,
+  event: {
+    type: "EServiceTemplateRiskAnalysisUpdated",
+    event_version: 2,
+    data: {
+      riskAnalysisId,
+      eserviceTemplate: toEServiceTemplateV2(eserviceTemplate),
+    },
+  },
+  correlationId,
+});
+
+export const toCreateEventEServiceTemplateDraftUpdated = (
+  streamId: string,
+  version: number,
+  eserviceTemplate: EServiceTemplate,
+  correlationId: CorrelationId
+): CreateEvent<EServiceTemplateEvent> => ({
+  streamId,
+  version,
+  event: {
+    type: "EServiceTemplateDraftUpdated",
+    event_version: 2,
+    data: {
+      eserviceTemplateId: eserviceTemplate.id,
```

Comment:
I have a doubt about this field id:  
it's the same that should be present in `eserviceTemplate` field? 
it's really necessary  ?
This doubt should be address in previous [PR](https://github.com/pagopa/interop-be-monorepo/pull/1395/) I leave a [comment](https://github.com/pagopa/interop-be-monorepo/pull/1395#discussion_r1925717151) about that.